OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -166,12 +166,21 @@ public CPos GetRandomBaseCenter()
 		readonly BaseBuilderQueueManager[] builders;
 		int currentBuilderIndex = 0;
 
+		readonly ActorIndex.NamesAndTrait<Building> refineryBuildings;
```

Comment:
Can we use an `ActorIndex.NamesAndTraitWithOwner` to pre-filter even further to just things owned by the player? (actors are removed/readded to the world when changing owner so the base index will be notified of owner changes).

Reply:
I didn't look into this originally as I expected it'd need more event handlers or extra juggling of some kind. But you're right, since the actor is removed and readded we can check for owner changes for free. I was able to convert almost all the indexes to filter on all three dimensions.